src/css_parser.rs

```diff
@@ -291,16 +269,31 @@ pub enum CssParsingError<'a> {
     UnsupportedCssKey(&'a str, &'a str),
 }
 
-impl_from!(CssBorderParseError, CssParsingError::CssBorderParseError);
-impl_from!(CssShadowParseError, CssParsingError::CssShadowParseError);
-impl_from!(CssColorParseError, CssParsingError::CssColorParseError);
-impl_from!(InvalidValueErr, CssParsingError::InvalidValueErr);
-impl_from!(PixelParseError, CssParsingError::PixelParseError);
-impl_from!(CssImageParseError, CssParsingError::CssImageParseError);
-impl_from!(CssFontFamilyParseError, CssParsingError::CssFontFamilyParseError);
-impl_from!(CssBackgroundParseError, CssParsingError::CssBackgroundParseError);
-impl_from!(CssBorderRadiusParseError, CssParsingError::CssBorderRadiusParseError);
-impl_from!(LayoutPaddingParseError, CssParsingError::PaddingParseError);
+impl_display!{ CssParsingError<'a>, {
+    CssBorderRadiusParseError(e) => format!("Invalid border-radius: {}", e),
+    CssBorderParseError(e) => format!("{}", e),
+    CssShadowParseError(e) => format!("Invalid shadow: \"{}\"", e),
```

Comment:
Wouldn't it be better to just do `format!("{}", e)`, to keep this consistent with the rest of the errors?

Reply:
Maybe.. I actually did that to make it consistent. 

This is the underlying `CssBorderRadiusParseError`:

```rust
#[derive(Debug, Clone, PartialEq, Eq)]
pub enum CssBorderRadiusParseError<'a> {
    TooManyValues(&'a str),
    PixelParseError(PixelParseError<'a>),
}

impl_display!{ CssBorderRadiusParseError<'a>, {
    TooManyValues(val) => format!("Too many values: \"{}\"", val),
    PixelParseError(e) => format!("{}", e)
}}
```

If I would have done `format!("{}", e)`, the valuable information that the error it is about a border would not have been included..

Reply:
Then do a `Invalid border property` on the `CssBorderParseError`. Otherwise, a border parse error will just display "Too many values" - which isn't helpful, since you don't know that it happened when parsing a `border` property.

Yeah, I sometimes reused errors because the parsing functions were essentially the same.

Reply:
I actually think that is perfectly clear that it is perfectly clear what the error is in this case:

```rust
CssBorderParseError(e) => format!("{}", e),
```

```rust
#[derive(Debug, Clone, PartialEq)]
pub enum CssBorderParseError<'a> {
    InvalidBorderStyle(InvalidValueErr<'a>),
    InvalidBorderDeclaration(&'a str),
    ThicknessParseError(PixelParseError<'a>),
    ColorParseError(CssColorParseError<'a>),
}

impl_display!{ CssBorderParseError<'a>, {
    InvalidBorderStyle(e) => format!("Invalid border style: {}", e.0),
    InvalidBorderDeclaration(e) => format!("Invalid border declaration: \"{}\"", e),
    ThicknessParseError(e) => format!("Invalid border thickness: {}", e),
    ColorParseError(e) => format!("Invalid border color: {}", e),
}}
```

But I could change it to this, if you think it improves consistency:

```rust
CssBorderParseError(e) => format!("Invalid border property: {}", e),
```

```rust
#[derive(Debug, Clone, PartialEq)]
pub enum CssBorderParseError<'a> {
    InvalidBorderStyle(InvalidValueErr<'a>),
    InvalidBorderDeclaration(&'a str),
    ThicknessParseError(PixelParseError<'a>),
    ColorParseError(CssColorParseError<'a>),
}

impl_display!{ CssBorderParseError<'a>, {
    InvalidBorderStyle(e) => format!("style: {}", e.0),
    InvalidBorderDeclaration(e) => format!("declaration: \"{}\"", e),
    ThicknessParseError(e) => format!("thickness: {}", e),
    ColorParseError(e) => format!("color: {}", e),
}}
```

Reply:
Yeah, then do the latter but maybe with an `Invalid style: {}` instead of just `style: {}` - more human-readable.